main.go

```diff
@@ -312,6 +312,18 @@ func main() {
 	}
 }
 
+func isValidCommit(commit string) bool {
+	if len(commit) != 40 {
+		return false
+	}
```

Comment:
I don't believe it's necessary to provide the full SHA. 

> Git is smart enough to figure out what commit you’re referring to if you provide the first few characters of the SHA-1 hash, as long as that partial hash is at least four characters long and unambiguous; that is, no other object in the object database can have a hash that begins with the same prefix.

https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection#_short_sha_1

Reply:
Can you check now, I have fixed the issue, also how to link the issue with the PR? I can't see the option to do so

Reply:
@rgmz

Reply:
Doing basic validation is helpful.

However, I think the only way to properly satisfy #3394 is to dispatch a command to Git. (e.g., https://stackoverflow.com/questions/18515488/how-to-check-if-the-commit-exists-in-a-git-repository-by-its-sha-1)

Reply:
Yeah, you are right, I have fixed by dispatching this command from the function.

Reply:
![Uploading Screenshot 2024-10-16 at 12.24.23 AM.png…]()